plugin/pkg/admission/alwayspullimages/admission.go

```diff
@@ -50,7 +51,7 @@ var _ admission.MutationInterface = &AlwaysPullImages{}
 var _ admission.ValidationInterface = &AlwaysPullImages{}
 
 // Admit makes an admission decision based on the request attributes
-func (a *AlwaysPullImages) Admit(attributes admission.Attributes) (err error) {
+func (a *AlwaysPullImages) Admit(_ request.Context, attributes admission.Attributes) (err error) {
```

Comment:
that's a deep and big change. Can't we attach the information to the admission.Attributes?

Reply:
Sure, we can attach `ctx` or `audit` to admission.Attributes.

Reply:
And also attach audit to authorizer attributes here?
https://github.com/kubernetes/kubernetes/blob/6b925ac24bb57c7cc1eab97d6b89a6f902dca30b/staging/src/k8s.io/apiserver/pkg/endpoints/filters/authorization.go#L49

Reply:
What exactly is `audit` that you have in mind, which type?

Reply:
>  What exactly is audit that you have in mind, which type?

Not very sure about this. But it seems to be something different from Verb, UserInfo.

But I am very happy to insert `audit` to admission.Attribute and  authorizer.Attribute. In fact I use such way in the beginning. This will make code much simpler.
:)

Reply:
Both admission and authorization interfaces are remoteable… how do we plan to get a non-serializeable parameter to a webhook authorizer and admission plugin?

Reply:
I would expect that a webhook can return an annotation map to persist in audit events.

Reply:
add a map to Admit() and Validate()s' return value?

Reply:
We also mutate the object "in" `admission.Attributes`. We should also store the audit annotation in `admission.Attributes` instead of complicating the function signatures.

Reply:
What about this:
https://github.com/kubernetes/kubernetes/pull/58143
I wrote it today.
I haven't meet a serialization promblem in #58143 yet.